packages/lucide/src/lucide.ts

```diff
@@ -17,10 +17,20 @@ const createIcons = ({ icons = {}, nameAttr = 'icon-name', attrs = {} } = {}) =>
   }
 
   const elementsToReplace = document.querySelectorAll(`[${nameAttr}]`);
-
   Array.from(elementsToReplace).forEach((element) =>
     replaceElement(element, { nameAttr, icons, attrs }),
   );
+
+  /** @todo: remove this block in v1.0 */
+  if (nameAttr === 'data-lucide') {
+    const deprecatedElements = document.querySelectorAll('[icon-name]');
+    if (deprecatedElements.length > 0) {
+      console.warn('Icons were found with the now deprecated icon-name attribute. These will still be replaced for backwards compatibility, but will no longer be supported in v1.0');
```

Comment:
Maybe mention our package name somewhere. So devs know where this warning is coming from.

Reply:
Oh, sure, also, we should add the new attribute name anyway.

Reply:
```suggestion
      console.warn('[Lucide] Some icons were found with the now deprecated icon-name attribute. These will still be replaced for backwards compatibility, but will no longer be supported in v1.0 and you should switch to data-lucide');
```

Reply:
perfecto!